packages/transaction-controller/src/helpers/IncomingTransactionHelper.ts

```diff
@@ -165,7 +166,8 @@ export class IncomingTransactionHelper {
       }
       if (!this.#updateTransactions) {
         remoteTransactions = remoteTransactions.filter(
-          (tx) => tx.txParams.to?.toLowerCase() === address.toLowerCase(),
+          (tx) =>
+            tx.txParams.to?.toLowerCase() === account.address.toLowerCase(),
```

Comment:
What about rewriting it like this:

```ts
const address = account.address.toLowerCase();
remoteTransactions = remoteTransactions.filter(
  (tx) =>
    tx.txParams.to?.toLowerCase() === address,
    ...
```

This way we only compute the lowercase address only once!